### PR TITLE
Lint using prettier + eslint

### PR DESCRIPTION
**Pull request by marcelomorgado**
These rules were needed to avoid errors on current App.js file:
``
"no-console": 0, 
"react/prop-types": 0,
``
I'm not sure about the best way how `npm run lint` should be, I've saw some examples that style/code script were splitted but I thought that is a bit verbose. What do you think?
Another point is about how files shoud be under linting. For now it's applying under App.js only.

### CODE REVIEW

File: demo/package.json [hunk 1]
```diff
@@ -9,7 +9,8 @@
     "android": "npx expo start --android",
     "ios": "npx expo start --ios",
     "eject": "npx expo eject",
-    "test": "npx jest"
+    "test": "npx jest",
+		"lint": "npx prettier --write App.js && npx eslint App.js"
```
Comment (pcowgill) [inline demo/package.json]: The indentation is weird compared with the rest of the file

**Comment by pcowgill** [comment]
We'll have a large directory of React Native components that we'll want linted too. I think basically any file ending in .js or .jsx should be linted. Does that make sense to you @marcelomorgado ?

**Comment by pcowgill** [comment]
If I remember correctly eslint now prefers words as opposed to numbers for their rule settings. We don't necessarily need to change the rules to match the current App.js - we could modify App.js as well. As long as the current settings still produce warnings though we can track that and work on it in a future PR.

**Comment by pcowgill** [comment]
Once we have CI configured I think it makes sense to run linting and tests as separate scripts that can fail independently and linking out to the results from GitHub PRs. But until we have that set up I think having lint run automatically when you run test is good.

File: demo/package.json [hunk 2]
```diff
@@ -20,7 +21,13 @@
     "tasit-sdk": "0.0.2"
   },
   "devDependencies": {
+    "babel-eslint": "^10.0.1",
     "babel-preset-expo": "^5.0.0",
+    "eslint": "^5.9.0",
+    "eslint-config-prettier": "^3.3.0",
+    "eslint-plugin-prettier": "^3.0.0",
```
Comment (pcowgill) [inline demo/package.json]: @marcelomorgado Are both the `eslint-config-prettier` and `eslint-plugin-prettier` required? They may very well be - I forget what the distinction is.

File: demo/package.json (hunk 1 above)
Comment (pcowgill) [reply demo/package.json]:
```suggestion
    "lint": "npx prettier --write App.js && npx eslint App.js"
```

**Comment by marcelomorgado** [comment]
> We'll have a large directory of React Native components that we'll want linted too. I think basically any file ending in .js or .jsx should be linted. Does that make sense to you @marcelomorgado ?

Yes, I'll do that.

File: demo/.eslintrc.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+	parser: "babel-eslint",
+	env: {
+		browser: true,
+		es6: true,
+		node: true,
+	},
+	plugins: ["react", "prettier", "react-native"],
+	extends: [
+		"eslint:recommended",
+		"plugin:react/recommended",
```
Comment (pcowgill) [inline demo/.eslintrc.js]:
```suggestion
		"plugin:react/recommended",
		"plugin:react-native/all",
```

File: demo/package.json [hunk 4]
```diff
@@ -9,7 +9,8 @@
     "android": "npx expo start --android",
     "ios": "npx expo start --ios",
     "eject": "npx expo eject",
-    "test": "npx jest"
+    "test": "npx jest",
+		"lint": "npx prettier --write '{*.js,**/*.js,*.jsx,**/*.jsx}' && npx eslint '{*.js,**/*.js,*.jsx,**/*.jsx}'"
```
Comment (pcowgill) [inline demo/package.json]:
```suggestion
    "lint": "npx prettier --write '{*.js,**/*.js,*.jsx,**/*.jsx}' && npx eslint '{*.js,**/*.js,*.jsx,**/*.jsx}'"
```

File: demo/package.json (hunk 2 above)
Comment (marcelomorgado) [reply demo/package.json]: @pcowgill `eslint-plugin-prettier` is used to run `prettier` by `eslint` and uses its as rules.
`eslint-config-prettier` avoid conflicts between both disabling `eslint` formatting rules.
More info: https://github.com/prettier/eslint-plugin-prettier#recommended-configuration

**Pull request merged**